From 937ec71b059734ebf585a56a768c68087062037c Mon Sep 17 00:00:00 2001
From: Cyprian Lech <clech@plume.com>
Date: Wed, 10 Jun 2020 12:19:08 +0000
Subject: [PATCH 7/7] bm: fix HWM configuration

In case cloud configures pref_5g == hwm,
block probe requests based only on hwm value.

The lwm value, even if configured, should be
ignored in this case.
---
 src/bm/src/bm_client.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/bm/src/bm_client.c b/src/bm/src/bm_client.c
index cc295e35..8b7b379c 100644
--- a/src/bm/src/bm_client.c
+++ b/src/bm/src/bm_client.c
@@ -221,7 +221,11 @@ bm_client_to_bsal_conf(bm_client_t *client, bsal_band_t band, bsal_client_config
         LOGT( "Client '%s': Setting hwm to '%hhu'",
                                 client->mac_addr, dest->rssi_probe_hwm );
 
-        dest->rssi_probe_lwm        = client->lwm;
+        if ( client->pref_5g == BM_CLIENT_5G_HWM ) {
+            dest->rssi_probe_lwm        = 0;
+        } else {
+            dest->rssi_probe_lwm        = client->lwm;
+        }
         dest->rssi_high_xing        = client->hwm;
         dest->rssi_inact_xing       = 0;
 
-- 
2.17.1

